src/main/java/repo/BookRepo.java

```diff
@@ -17,12 +17,34 @@ public void addBook(Book book) {
         }
     }
 
-    public Book getBook(String name,int isbn) {
+    public ArrayList<Book> findBooks(Book book) {
+        ArrayList<Book> result = new ArrayList<>();
+        for (Book i : books) {
+            if (book.getName().equals(i.getName()) && book.getAuthor().equals(i.getAuthor())) {
+                result.add(i);
+            }
+        }
+        return result;
+    }
+
+    public Book getBook(String name, String author) {
         for (Book book : books) {
-            if (book.equals(name) && book.getIsbn()== isbn) {
+
+            if (book.getName().equals(name) && book.getAuthor().equals(author)) {
                 return book;
             }
         }
         return null;
     }
+
+    public Book getBookReservation(String name, String author, int isbn) {
+        for (Book i : books) {
+            if (i.getName().equals(name) && i.getAuthor().equals(author) & i.getIsbn() == isbn) {
```

Comment:
Why do you have two methods that do the same thing?
- `getBook` returns a book given a name and an author
- `getBookReservation` returns a book given the name, author, and an ISBN.

This violates one of the principles of software called DRY:
Don't Repeat Yourself

I suggest the following:
- One method that returns a list of books given a name and an author
- One method that returns a single book given only the ISBN


Also, this method is called `getBookReservation` and lives under the `BookRepo`, that makes no sense, why there is something from reservations living in the books repo?